packages/frontend/src/components/MetricQueryData/UnderlyingDataModal.tsx

```diff
@@ -8,20 +8,26 @@ const UnderlyingDataModal: FC = () => {
         useMetricQueryDataContext();
 
     return (
-        <Dialog
-            isOpen={isUnderlyingDataModalOpen}
+        <Modal
+            centered
+            opened={isUnderlyingDataModalOpen}
             onClose={closeUnderlyingDataModal}
-            lazy
-            title={`View underlying data`}
-            style={{
-                width: '90%',
-                height: '90vh',
-                minHeight: '400px',
-                minWidth: '500px',
+            title={<Title order={5}>View underlying data</Title>}
+            size="auto"
+            styles={{
+                body: {
+                    height: '100%',
+                },
+                content: {
+                    minWidth: '100dvh',
+                    minHeight: 'calc(100dvh - (4rem * 2))',
+                    height: 'calc(100dvh - (10rem * 2))',
```

Comment:
whats happeniiiing 😱 
can we have 8 rem and 20 rem?
do we need rems at all?
what's 100dvh I have no idea

Reply:
I know... this is not ideal. Let's catch up because I need your CSS expertise 😆 

definition: 
dvw, dvh	1% of the [dynamic viewport](https://developer.mozilla.org/en-US/docs/Web/CSS/length#relative_length_units_based_on_viewport)'s width and height, respectively.

from the link above ^:
```
The dynamic viewport size allows the content you design to fit exactly within the viewport, irrespective of the presence of dynamic browser interfaces.
```